src/verifySignedMessage.js

```diff
@@ -0,0 +1,141 @@
+  }
+
+  if (functionName !== 'metaDelegateCall') {
+    throw new Error(`Unsupported functionName \'${functionName}\', only \'metaDelegateCall\' supported`)
+  }
+
+  if (!signedParams || signedParams.length === 0 || !signedParams[1]) {
+    throw new Error('signedParams not provided in signedMessage')
+  }
+
+  if (!EIP712TypedData) {
+    throw new Error('EIP712TypedData not provided in signedMessage')
+  }
+
+  const callData = signedParams[1].callData
+  if (!callData) {
+    throw new Error('CallData not provided in signedMessage Params')
+  }
+
+  const recoveredSigner = recoverSigner({ signature, typedDataHash: message})
+  if (recoveredSigner !== signer) {
+    throw new Error(`Provided Signer ${signer} does not match Signer ${recoveredSigner} in Signed Message`)
+  }
+
+  const computedAccountAddress = accountFromOwner(signer)
+  if (toChecksumAddress(computedAccountAddress) !== toChecksumAddress(accountAddress)) {
+    throw new Error(`Account Address ${accountAddress} does not match Computed Address ${computedAccountAddress}`)
+  }
+
+  const paramValues = signedParams.map(p => p.value)
+  const { typedDataHash } = typedDataEIP712({ 
+    accountVersion: EIP712TypedData.domain.version, 
+    chainId: EIP712TypedData.domain.chainId,
+    accountAddress,
+    functionName,
+    paramTypes: metaDelegateCallParamTypes,
+    params: paramValues
+  })
+
+  if (typedDataHash !== message) {
+    throw new Error(`Provided signed message hash ${message} does not match recovered hash ${typedDataHash}`)
+  }
+
+  const eip712To = EIP712TypedData.value.to
+  const eip712Data = EIP712TypedData.value.data
+  let toValueExists = false
+  let dataValueExists = false
+  for (let i=0; i<paramValues.length; i++) {
+    if (paramValues[i] === eip712To) {
+      toValueExists = true
+    }
+    if (paramValues[i] === eip712Data) {
+      dataValueExists = true
+    }
+  }
+
+  if (!toValueExists || !dataValueExists) {
+    throw new Error(`Signed message params do not match EIP712TypedData params`)
+  }
+
+
+  const callDataFunctionName = callData.functionName
+  if (!callDataFunctionName) {
+    throw new Error('Function name not provided in calldata')
+  }
+  const callDataParams = callData.params
+  if (!callDataParams) {
+    throw new Error('Params not provided in calldata')
+  }
+
+  const callDataParamTypes = callDataParams.map(p => p.type)
+  const callDataParamNames = callDataParams.map(p => p.name)
+
+  if (!callDataParamTypes) {
+    throw new Error('Param Types not provided in calldata')
+  }
+
+  if (!callDataParamNames) {
+    throw new Error('Param Names not provided in calldata')
+  }
+
+  let callDataParamNameTypes = []
+  for (let i=0; i< callDataParamTypes.length; i++) {
+    const callDataParamNameType = {name: callDataParamNames[i], type: callDataParamTypes[i]}
+    callDataParamNameTypes.push(callDataParamNameType)
+  }
+  const callDataParamValues = callDataParams.map(p => p.value)
+  callDataParamValues.pop()
```

Comment:
so this will work with the limit swap calls, because there are always two "unsigned" params at the end.. but it might be better to do this for all params in `callDataParams` that don't have values defined. There could be any 0..n of these